src/Umbraco.Web/Editors/MediaController.cs

```diff
@@ -278,6 +281,31 @@ public IEnumerable<MediaItemDisplay> GetByIds([FromUri]int[] ids)
         }
 
         /// <summary>
+        /// Searches media and returns a paged result
+        /// </summary>
+        /// <param name="query"></param>
+        /// <param name="pageNumber"></param>
+        /// <param name="pageSize"></param>
+        /// <param name="searchFrom"></param>
+        /// <returns></returns>
+        [HttpGet]
+        [FilterAllowedOutgoingMedia(typeof(IEnumerable<ContentItemBasic<ContentPropertyBasic, IMedia>>), "Items")]

```

Comment:
@madsrasmussen  + @emilwangaa Pro tip! The EntityController and EntityService, when returning Media items will contain all of the media item's properties in it's `AdditionalData` (in c#) or `metaData` (in JS) collections, so you can still access the media properties.

Looks like the `umbTreeSearchBox` was added which does an Examine search a while back when the directives changed. This is unfortunate because I've said this a few times that I don't want the pickers to use Examine Searches but I guess that was hacked in there without my knowing. In any case, it is what it is, I'd much rather fix that to use DB searches with a new method like EntityController.GetDescendants. I am in fact making an EntityController.GetChildren method right now which would be the same type of thing so is easy to do. Perhaps you can use this instead.

In the meantime, I'm leaving this in review, I'll remove all of the new code duplication that exists in the MediaController (ExamineSearch) (this is 100% unnecessary). We can keep MediaController.Search if we want, in which case we can add that to ContentController too. I'll also add EntityController.GetDescendants and EntityController.GetChildren since we need that for http://issues.umbraco.org/issue/U4-9470

So another thing to know about EntityController vs Media/Content Controller - The EntityController returns generic entities for all types, generally without any of it's properties. Media is slightly different because it's properties are put into the AdditionalData/metaData properties because for the most part all media user properties need to be known since that is where the media info is actually stored. The Entity returned from EntityController also tells you if the item HasChildren, the Content/Media Controllers *do not*. This is because the Content and Media objects returned from those controllers are complex models including all properties, content type, property type, etc... information. It is far to inefficient to make these models more complicated by also including if they have children.